### ISSUE TITLE
Focus follows mouse not working for other applications

### ISSUE THREAD

**Issue by dakra** [ch11ng/exwm#340]
I have set
```(setq mouse-autoselect-window t)
(setq focus-follows-mouse t)
```
and focus follows the mouse nicely between Emacs frames but if I
go e.g. from Emacs to Firefox (or any other "external" application) it doesn't work.
From external Application to Emacs it's even a bit more wired since the first Emacs frame where the mouse enters doesn't get activated but hovering then over a second different emacs window it gets focused.

I use latest exwm and emacs from git.

**Comment by ch11ng** [comment]
This is not supported yet and the behavior is largely unpredictable.  I skipped this part for the simplicity of the input focus model.  But if you really have use for this, I can consider implementing it.  Please make a list of typical workflows involving this option.

**Comment by dakra** [comment]
I'm just so used to select windows this way that
having to click on stuff is annoying for me.
I'm surprised nobody else is bothered by it since
I would have thought it's the default behaviour of most(?)
X11 window managers.

I don't want it because I want to use the mouse,
for me it's just a fast way to select windows.

I work most of the time on my laptop and the trackpad
is right there where my thumb is, so I can hack in
one window and with a quick thumb swipe I can
basically switch to any window I want without having
to move my hand much.

I do have windmove and ace-window set up and it's also
fast to move that way but imho just swiping the mouse is
even faster and also in my muscle memory.

That's basically the only usecase I have ;)
Adding to the confusion is that it works nicely in Emacs windows
and sometime from other -> Emacs but never from Emacs to other apps.

**Comment by QiangF** [comment]
It would be a nice feature to interact with floating windows.

**Comment by ch11ng** [comment]
This feature is in the plan.  Stay tuned.

**Comment by dakra** [comment]
@ch11ng nice :)

One thing I didn't mention is that I preferably would like the same mouse-warping behaviour then i3wm. And that is that changing windows (with keyboard) to the same output leaves the mouse where it is but changing from one output to another it warps the mouse.
https://i3wm.org/docs/userguide.html#_mouse_warping

**Comment by ch11ng** [comment]
I've pushed an implementation.  Please give it a try.

**Comment by dakra** [comment]
@ch11ng I tested it now a bit and works nicely. Thanks.

**Comment by ptxmac** [comment]
This is awesome! - I'm slowly recreating my current i3 setup to exwm, keep up the good work!
